### PR TITLE
nixos/geoclue2: wait for network to be up when wifi provider is enabled

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Description of changes

The wifi provider can't work without it after all.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/desktops/geoclue2.nix [hunk 1]
```diff
@@ -216,6 +216,11 @@ in
         # we can't be part of a system service, and the agent should
         # be okay with the main service coming and going
         wantedBy = [ "default.target" ];
+        after = lib.optionals cfg.enableWifi [
+          "NetworkManager-wait-online.service"
+          "network.target"
+          "systemd-networkd-wait-online.service"
+        ];
```
Comment (winterqt) [inline nixos/modules/services/desktops/geoclue2.nix:223]: Wouldn't `network-online.target` suffice here?

Comment (lovesegfault) [reply nixos/modules/services/desktops/geoclue2.nix]: Indeed :)

**Review by winterqt — APPROVED** [review]
Waiting on OfBorg.

**Comment by winterqt** [comment]
> The wifi provider can't work without it after all.

If network *isn't* available, it doesn't blow up, right? Like, should this be required, or does after suffice? I think after is probably fine, but you know more about the software than me, so just making sure.

**Comment by lovesegfault** [comment]
After is enough, AFAICT

**Comment by m-bdf** [comment]
`network-online.target` doesn't exist in systemd user instances, so this does nothing.
There actually seems to be no way to ensure network is up before starting a user unit :confused:

**Pull request merged**